includes/filename.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+// Exit if accessed directly
+if(!defined('ABSPATH')){
+	exit;
+}
+
+/**
+ * Default functions
+ */
+class Fiber_Admin_Filename{
+	public function __construct(){
+		// Auto Convert File Name
+		if(fiad_get_miscellaneous_option('auto_convert_file_name')){
+			add_filter('sanitize_file_name', [$this, 'fiad_auto_convert_file_name'], 10, 2);
+		}
+	}
+	
+	// Auto Convert File Name
+	public function fiad_auto_convert_file_name($filename, $filename_raw){
+		//handle urlencode case
+		$sanitized_filename = $filename;
+		$url_decode_raw     = urldecode($filename_raw);
+		$sanitized_filename = str_split($sanitized_filename);
+		if($url_decode_raw != $filename_raw){
+			$count = 0;
+			foreach (str_split($url_decode_raw) as $index => $char) {
+				if ($char === ' ') {
+					if ($count == 0) {
+						$sanitized_filename[$index-1] = '';
+						$sanitized_filename[$index] = '-';
+					} else {
+						$sanitized_filename[$index] = '';
+						$sanitized_filename[$index+1] = '-';
+					}
+					$count++;
+				}
+			}
+		}
+		$sanitized_filename = implode('',$sanitized_filename);
+		
+		//special char case
```

Comment:
I think these codes should be separated into another function to make it easier to modify in the future.